torch/nn/functional.py

```diff
@@ -4568,7 +4568,7 @@ def interpolate(  # noqa: F811
         # Two levels are necessary to prevent TorchScript from touching
         # are_deterministic_algorithms_enabled.
         if not torch.jit.is_scripting():
-            if torch.are_deterministic_algorithms_enabled() and input.is_cuda:
+            if torch.are_deterministic_algorithms_enabled() and (input.is_cuda or input.is_xpu):
```

Comment:
1. Is this check needed at the first place? I mean I wonder whether `input.is_cuda()` condition should be just removed instead?
2. If it's really needed, then why this check is added here and is not added here (this 2nd place is in `pad()`):
https://github.com/pytorch/pytorch/blob/0d0305e774fc40833047bf95fbb49f1a35615f2a/torch/nn/functional.py#L5084

Reply:
> 1. Is this check needed at the first place? I mean I wonder whether `input.is_cuda()` condition should be just removed instead?
> 2. If it's really needed, then why this check is added here and is not added here (this 2nd place is in `pad()`):
>    https://github.com/pytorch/pytorch/blob/0d0305e774fc40833047bf95fbb49f1a35615f2a/torch/nn/functional.py#L5084

1, For other backend, like cpu, whose kernel is deterministic implementation. So there is not need to call decompose implementation, which has slow performance.
2, yes, we also need the 2nd, I am working on interpolate, and haven't validate pad related functionality. Would you like a sperate pr or, I just add pad change in this pr ?

Reply:
> Would you like a sperate pr or, I just add pad change in this pr?

I suggest to have this in this PR.